packages/expo-apple-authentication/src/AppleAuthentication.ts

```diff
@@ -0,0 +1,107 @@
+import { EventEmitter, Subscription, UnavailabilityError } from '@unimodules/core';
+import ExpoAppleAuthentication from './ExpoAppleAuthentication';
+
+import {
+  RequestOptions,
+  Credential,
+  CredentialState,
+  Operation,
+  RevokeListener,
+} from './AppleAuthentication.types';
+
+/**
+ * A method which returns a Promise which resolves to a boolean if you are able to perform a Sign In with Apple.
+ * Generally users need to be on iOS 13+.
+ */
+export async function isAvailableAsync(): Promise<boolean> {
+  if (!ExpoAppleAuthentication || !ExpoAppleAuthentication.isAvailableAsync) {
+    return false;
+  }
+  return ExpoAppleAuthentication.isAvailableAsync();
+}
+
+/**
+ * Perform a Sign In with Apple request with the given `RequestOptions`.
+ * The method will return a Promise which will resolve to a `Credential` on success.
+ * You should make sure you include error handling.
+ *
+ * @example
+ * ```ts
+ * import * as AppleAuthentication from "expo-apple-authentication";
+ *
+ * AppleAuthentication.requestAsync({
+ *   requestedScopes: [
+ *     AppleAuthentication.Scope.FullName,
+ *     AppleAuthentication.Scope.Email,
+ *   ]
+ * }).then(credentials => {
+ *   // Handle successful authenticated
+ * }).catch(error => {
+ *   // Handle authentication errors
+ * })
+ * ```
+ */
+export async function requestAsync(options: RequestOptions): Promise<Credential> {
```

Comment:
+1 to the idea of defining more precise methods like (login/logout/refresh)Async in JS. This lets each JS method take different arguments (LoginOptions, LogoutOptions, ...) and then they all could call requestAsync internally, and we can write Jest tests to test the unit of the login/logout/refresh methods and their integration with the requestAsync interface.